frontend/src/pages/result/VoteResult.tsx

```diff
@@ -1,51 +1,87 @@
-import * as React from 'react';
 import Tabs from '@mui/material/Tabs';
 import Tab from '@mui/material/Tab';
 import Box from '@mui/material/Box';
 import DisplayMenus from "../../components/general/DisplayMenus";
-import {TastyDateItem} from "../../models/result/TastyDateItem";
 import VoteTimeTable from '../../components/result/VoteTimeTable';
-import {a11yProps, TabPanel} from "./ResultTab";
-import {useState} from "react";
+import {TabPanel} from "./TabPanelFunctions";
+import React, {useContext, useEffect, useState} from "react";
 import VoteRestaurantTable from "../../components/result/VoteRestaurantTable";
+import {TextField} from "@mui/material";
+import {useParams} from "react-router-dom";
+import {TastyDateItem} from "../../models/result/TastyDateItem";
+import {AuthContext} from "../../context/AuthProvider";
+import {getTastyDateItemById} from "../../service/tastydate-api-service";
+import Invitationlink from "../../components/result/Invitationlink";
 
 
-interface VoteResultProps {
-    transferSettingsItem: TastyDateItem
-    setTransferSettingsItem: Function
-}
+export default function VoteResult() {
+
+    const {tastyDateId}:{tastyDateId?:string} = useParams();
+
+    const {token}=useContext(AuthContext)
+
+    const [tastyDateItemForVote, setTastyDateItemForVote] = useState<TastyDateItem>();
 
-export default function VoteResult({transferSettingsItem, setTransferSettingsItem}:VoteResultProps) {
-    const [value, setValue] = React.useState(0);
+    const getTastyDateItem = () => {
+        if (tastyDateId) {
+        getTastyDateItemById(tastyDateId, token)
+            .then((response)=>setTastyDateItemForVote(response.data))
+        }
+    }
+
+    useEffect(()=>{
+        getTastyDateItem()
+        //eslint-disable-next-line
+    },[])
+
+
+    const [voteType, setVoteType] = useState(0);
     const [userName, setUserName] = useState<string>("");
-    const [tempName, setTempName] = useState<string>("");
+    const [checkIfNameConfirmed, setCheckIfNameConfirmed] = useState<boolean>(false);
 
     const handleChange = (event: React.SyntheticEvent, newValue: number) => {
-        setValue(newValue);
+        setVoteType(newValue);
     };
 
+    if (typeof tastyDateItemForVote === "object") {
```

Comment:
Why do you check for type? Is there a possiblity that you recive something else than a object? I would check for undefined